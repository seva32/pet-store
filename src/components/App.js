import React from "react";
import { Switch, BrowserRouter as Router, Route } from "react-router-dom";
import Home from "./Home";
// this import should enable import transpilation
import "react-imported-component/macro";
import { AppProvider } from "../context/AppProvider";

// splitting code
// import "./load.async-imports";
// import Load from "./load";
// const AsyncDynamicPAge = Load(() =>
//   import(/* webpackChunkName:'DynamicPage' */ "./DynamicPage")
// );
// const AsyncNoMatch = Load(() =>
//   import(/* webpackChunkName:'NoMatch' */ "./NoMatch")
// );

// alltoghether
import importedComponent from "react-imported-component";
import Loading from "./Loading";
const AsyncDynamicPAge = importedComponent(
  () => import(/* webpackChunkName:'DynamicPage' */ "./DynamicPage"),
  {
    LoadingComponent: Loading,
  }
);
const AsyncNoMatch = importedComponent(
  () => import(/* webpackChunkName:'NoMatch' */ "./NoMatch"),
  {
    LoadingComponent: Loading,
  }
);

const App = () => {
  return (
    <Router>
      <AppProvider>
        <div>
          <Switch>
            <Route exact path="/" component={Home} />
            <Route exact path="/dynamic" component={AsyncDynamicPAge} />
            <Route component={AsyncNoMatch} />
          </Switch>
        </div>
      </AppProvider>
    </Router>
  );
};

export default App;
