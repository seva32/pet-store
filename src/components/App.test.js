import React from "react";
import App from "./App";
//import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });
import { shallow } from "enzyme";

it("renders correctly enzyme", () => {
  const wrapper = shallow(<App />);
  expect(toJson(wrapper)).toMatchSnapshot();
}); // We just need to press "w" to activate watch mode then press "u" to update the snapshot.
