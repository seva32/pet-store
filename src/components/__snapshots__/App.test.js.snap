// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly enzyme 1`] = `
<BrowserRouter>
  <div>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "done": Promise {},
            "preload": [Function],
            "render": [Function],
          }
        }
        exact={true}
        path="/dynamic"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "done": Promise {},
            "preload": [Function],
            "render": [Function],
          }
        }
      />
    </Switch>
  </div>
</BrowserRouter>
`;
